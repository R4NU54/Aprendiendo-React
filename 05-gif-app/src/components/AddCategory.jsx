import { useState } from 'react';

// eslint-disable-next-line react/prop-types
export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (!inputValue) return;
    setInputValue('');
    onNewCategory(inputValue);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type='text'
        placeholder='Search...'
        value={inputValue}
        onChange={handleInputChange}
      />
      {/* <button type='submit'>Search</button> */}
    </form>
  );
};
