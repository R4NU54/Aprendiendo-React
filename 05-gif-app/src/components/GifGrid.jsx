import { GiftItem } from './GiftItem';

import { useFetchGifs } from '../hooks/useFetchGifs';

// eslint-disable-next-line react/prop-types
export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <section className='section-search'>
      <h3>{category}</h3>
      {isLoading && <p>Loading...</p>}
      <div className='card-grid'>
        {images.map((images) => (
          <GiftItem
            key={images.id}
            {...images}
          />
        ))}
      </div>
    </section>
  );
};
