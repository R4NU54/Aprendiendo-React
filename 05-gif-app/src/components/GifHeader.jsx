import image from '../../GifApp.webp';

export const GifHeader = () => {
  return (
    <header className='header'>
      <img
        alt='GifApp'
        className='gif-app'
        src={image}
      />
      <h1>React Gif App</h1>
    </header>
  );
};
