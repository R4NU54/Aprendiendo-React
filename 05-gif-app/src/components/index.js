export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';
export * from './GifHeader.jsx';
