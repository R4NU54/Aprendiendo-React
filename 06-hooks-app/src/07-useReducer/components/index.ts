export * from './todoAdd';
export * from './todoList';
export * from './todoItem';
