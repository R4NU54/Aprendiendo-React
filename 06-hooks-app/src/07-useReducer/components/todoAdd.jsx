import { useForm } from '../../hooks/useForm';
export const TodoAdd = ({ onNewTodoAdd }) => {
  const { formState, handleChange, onResetForm, description } = useForm({
    description: '',
  });

  const handleSubmit = e => {
    e.preventDefault();
    if (description.length <= 1) return;

    const idTodo = crypto.randomUUID();

    const newTodo = {
      id: idTodo,
      description: description,
      done: false,
    };

    onNewTodoAdd(newTodo);
    onResetForm();
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        name="description"
        placeholder="Agrega todo..."
        autoComplete="off"
        className="form-control"
        value={description}
        onChange={handleChange}
      />
      <button type="submit" className="btn btn-outline-primary mt-1 btn-block">
        Agregar
      </button>
    </form>
  );
};
