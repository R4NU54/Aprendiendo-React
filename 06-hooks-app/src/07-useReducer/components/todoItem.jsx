export const TodoItem = ({ todo, onDeleteTodo, onDoneTodo }) => {
  return (
    <li className="list-group-item d-flex justify-content-between p-2">
      <span aria-label="span" className="align-self-center" onClick={() => onDoneTodo(todo.id)}>
        {todo.done ? <s>{todo.description}</s> : todo.description}
      </span>
      <button className="btn btn-danger" onClick={() => onDeleteTodo(todo.id)}>
        Borrar
      </button>
    </li>
  );
};
