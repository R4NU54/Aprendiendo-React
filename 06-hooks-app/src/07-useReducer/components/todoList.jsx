import { TodoItem } from './todoItem';

export const TodoList = ({ todos = [], onDeleteTodo, onDoneTodo }) => {
  return (
    <ul className="d-flex gap-2 list-group">
      {todos.map(todo => (
        <TodoItem key={todo.id} todo={todo} onDeleteTodo={onDeleteTodo} onDoneTodo={onDoneTodo} />
      ))}
    </ul>
  );
};
