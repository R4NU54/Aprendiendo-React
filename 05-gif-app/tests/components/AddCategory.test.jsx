import { render, screen, fireEvent } from '@testing-library/react';

import { AddCategory } from '../../src/components/AddCategory';

describe('Pruebas en el componente AddCategory', () => {
  test('Debe ser como el snapshot', () => {
    const { container } = render(<AddCategory onNewCategory={() => {}} />);

    expect(container).toMatchSnapshot();
  });

  test('Debe cambiar el valor del input', () => {
    render(<AddCategory onNewCategory={() => {}} />);
    const input = screen.getByRole('textbox');

    fireEvent.input(input, { target: { value: 'Hola mundo' } });
    expect(input.value).toBe('Hola mundo');
  });

  test('debe llamar onNewCategory si el input tienen un valor', () => {
    const inputValue = 'Hola mundo!';
    const onNewCategory = jest.fn();

    render(<AddCategory onNewCategory={onNewCategory} />);

    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form');

    fireEvent.input(input, { target: { value: inputValue } });
    fireEvent.submit(form);

    expect(input.value).toBe('');
    expect(onNewCategory).toHaveBeenCalledTimes(1);
    expect(onNewCategory).toHaveBeenCalledWith(inputValue);
  });

  test('no debe ser llamar al onNewCategory si el input esta vacio', () => {
    const onNewCategory = jest.fn();

    render(<AddCategory onNewCategory={onNewCategory} />);

    const form = screen.getByRole('form');

    fireEvent.submit(form);

    expect(onNewCategory).not.toHaveBeenCalled();
  });
});
