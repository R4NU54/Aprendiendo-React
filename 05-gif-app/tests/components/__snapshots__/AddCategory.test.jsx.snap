// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory Debe ser como el snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search..."
      type="text"
      value=""
    />
  </form>
</div>
`;
